hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -497,6 +532,7 @@ private void unlock(Optional<Long> lockId) {
     if (lockId.isPresent()) {
       try {
         doUnlock(lockId.get());
+        tableLocksById.remove(lockId.get());
```

Comment:
If there is a problem with unlock, this could cause a memory leak

Reply:
Thanks. I've fixed it.